### PR TITLE
Amls 4945 2

### PR DESCRIPTION
**Pull request by ajchilts**
Create audit event for notification read fails. 

## Related / Dependant PRs?

N/A

## How Has This Been Tested?

Unit + Regression. 

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [ ] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: test/audit/MockAudit.scala [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
 package audit
 
-import uk.gov.hmrc.play.audit.model.{Audit, DataEvent}
+import uk.gov.hmrc.play.audit.model.{Audit, DataEvent, ExtendedDataEvent}
 
 // This only exists because of the difficulty in mocking
 // the `sendDataEvent` method of the `Audit` class
 // scalastyle:off null
-object MockAudit extends Audit("mockApp", null) {
+class MockAudit extends Audit("mockApp", null) {
```
Comment (drewmiley) [inline test/audit/MockAudit.scala]: This is so this `mock[MockAudit]` ?

File: app/connectors/ViewNotificationConnector.scala [hunk 2]
```diff
@@ -61,10 +61,13 @@ class ViewNotificationConnector @Inject()(
       case r@status(s) =>
         metrics.failed(API11)
         Logger.warn(s"$prefix - Failure response: $s")
+        val httpEx = HttpStatusException(s, Option(r.body))
```
Comment (drewmiley) [inline app/connectors/ViewNotificationConnector.scala]: Worth adding type to this?

File: app/audit/ViewNotificationEvent.scala [hunk 3]
```diff
@@ -41,3 +42,22 @@ object ViewNotificationEvent {
...
+   resW: Writes[NotificationResponse]
+  ): DataEvent =
+    DataEvent(
+      auditSource = AuditHelper.appName,
+      auditType = "viewNotificationEventFailed",
+      tags = hc.toAuditTags("Get Notification Failed", "N/A"),
+      detail = hc.toAuditDetails() ++ Map(
+        "amlsRegNo" -> amlsRegistrationNumber,
+        "contactNumber" -> contactNumber,
+        "reason" -> Json.toJson(ex.body.getOrElse("No body found")).toString
```
Comment (drewmiley) [inline app/audit/ViewNotificationEvent.scala]: Out of curiosity, why do we need `toString` ?

Comment (ajchilts) [reply app/audit/ViewNotificationEvent.scala]: The DataEvent takes params of type string. So if we remove the .toString() we encounter a compiler error;

found   : scala.collection.immutable.Map[String,Object]
 required: Map[String,String]

Comment (drewmiley) [reply app/audit/ViewNotificationEvent.scala]: 👍

**Comment by ajchilts** [comment]
Acceptance tests passing for Notifications. [/]

**Pull request merged**